swarm/storage/localstore.go

```diff
@@ -205,8 +205,12 @@ func (ls *LocalStore) Migrate() error {
 	if schema != CurrentDbSchema {
 		// run migrations
 
-		if schema == "" {
```

Comment:
I find this confusing.

We should have an if-block for every type of dbschema that is possible on users machines.

It is possible that a user has not updated their Swarm process and are still on schema == "", it is possible that they are also on schema purity and it is possible that they have updated to halloween - this makes it 3 blocks.

Having nested if checks for this is not ideal.